src/boilerplate/orchestration/javascript/raw/toOrchestration.ts

```diff
@@ -760,13 +772,33 @@ export const OrchestrationCodeBoilerPlate: any = (node: any) => {
       return {
         statements: [
           `\n\n// Send transaction to the blockchain:
-          \nconst tx = await instance.methods
-          .${node.functionName}(${lines}${params[0][0]} ${params[0][1]} ${params[0][2]} ${params[0][3]} ${params[0][4]} ${params[0][5]} proof)
-          .send({
-              from: config.web3.options.defaultAccount,
-              gas: config.web3.options.defaultGas,
-              value: msgValue,
-            });\n`,
+          \nconst txData = await instance.methods
+          .${node.functionName}(${lines}${params[0][0]} ${params[0][1]} ${params[0][2]} ${params[0][3]} ${params[0][4]} ${params[0][5]} proof).encodeABI();
+          \n	let txParams = {
+            from: config.web3.options.defaultAccount,
+            to: contractAddr,
+            gas: config.web3.options.defaultGas,
+            gasPrice: config.web3.options.defaultGasPrice,
+            data: txData,
+            chainId: await web3.eth.net.getId(),
+            };
+            \n 	const key = config.web3.key;
+            \n 	const signed = await web3.eth.accounts.signTransaction(txParams, key);
+            \n 	const sendTxn = await web3.eth.sendSignedTransaction(signed.rawTransaction);
+            \n  let tx = await instance.getPastEvents("NewLeaves");
+            \n tx = tx[0];\n
+            let encEvent = '';
```

Comment:
Would it be possible to just handle events outside of the main orchestration file? And just have code in `test` and `api_services` to extract?
In some cases we won't have any encrypted data (e.g. updating your own whole state) so we don't need it in every function - I realise this does mean we have to get the contract instance in those outside files, but I think that's ok.

Reply:
Just to add, this does cause an error in `api_services` since we try and extract the events there from the returned tx

Reply:
Thanks for noting. For `test` , I have wrapped the events inside a try catch to handle scenarios where we dont expect encrypted data.

Reply:
- Events handling outside main orchestration will require starlight end users to handle events which might not be best approach. We can handle empty events with a try catch and throw an empty response in car no event is emitted.
- Modified `registerKey` in `contract.mjs` with new signing method. Thanks for noting